backend/common/src/test/java/ai/verta/modeldb/common/futures/InternalFutureTest.java

```diff
@@ -14,35 +14,35 @@ class InternalFutureTest {
   void composition_failsFast() {
     Executor executor = MoreExecutors.directExecutor();
     InternalFuture<String> testFuture =
-        InternalFuture.completedInternalFuture("cheese")
+        InternalFuture.withExecutor(executor)
+            .completedInternalFuture("cheese")
             .thenApply(
                 s1 -> {
                   throw new RuntimeException("borken");
-                },
-                executor)
+                })
             .thenCompose(
                 s -> {
                   Assertions.fail("should never execute the next stage");
                   return InternalFuture.failedStage(new RuntimeException("broken"));
-                },
-                executor);
+                });
 
     assertThatThrownBy(testFuture::get).getRootCause().hasMessage("borken");
   }
 
   @Test
   void thenSupply() {
     AtomicBoolean firstWasCalled = new AtomicBoolean();
-    Executor executor = MoreExecutors.directExecutor();
-    InternalFuture<Void> testFuture =
-        InternalFuture.supplyAsync(
+    final Executor executor = MoreExecutors.directExecutor();
+    final InternalFutureWithDefaultExecutor.FactoryWithExecutor factory =
```

Comment:
nit: these `final` modifiers don't add much value, and clutter up the code.
```suggestion
   Executor executor = MoreExecutors.directExecutor();
   InternalFutureWithDefaultExecutor.FactoryWithExecutor factory =
```